components/DateTime.js

```diff
@@ -12,9 +13,9 @@ class DateTime extends React.Component {
   render() {
     const props = this.props;
     const { date, timezone } = props;
-    const value = momentTimezone.tz(new Date(date), timezone);
+    const value = dayjs(new Date(date)).tz(timezone);
```

Comment:
Yes, it's flaky and I think the issue is with `dayjs` `timezone` plugin. It works fine without it. 
I'm not sure how this should be fixed.